jconstraints-core/src/main/java/gov/nasa/jpf/constraints/smtlibUtility/smtconverter/SMTLibExportVisitor.java

```diff
@@ -58,16 +59,33 @@
 public class SMTLibExportVisitor extends AbstractExpressionVisitor<Void, Void> {
 
   private final SMTLibExportGenContext ctx;
+  private SMTLibExportVisitorConfig config;
+
+  public SMTLibExportVisitor(SMTLibExportGenContext ctx, SMTLibExportVisitorConfig config) {
+    this(ctx);
+    this.config = config;
+  }
 
   public SMTLibExportVisitor(SMTLibExportGenContext ctx) {
     this.ctx = ctx;
+    this.config = new SMTLibExportVisitorConfig();
   }
```

Comment:
This creates an extra object; we could flip the constructor chain:
```suggestion
  public SMTLibExportVisitor(SMTLibExportGenContext ctx, SMTLibExportVisitorConfig config) {
    this.ctx = ctx;
    this.config = config;
  }

  public SMTLibExportVisitor(SMTLibExportGenContext ctx) {
    this(ctx, new SMTLibExportVisitorConfig());
  }
```